src/main/java/dk/kb/datahandler/oai/OaiResponseFiltering.java

```diff
@@ -43,17 +43,25 @@ public static void addToStorageWithoutFiltering(OaiResponse response, DsStorageA
      */
     public static void addToStorageWithPvicaFiltering(OaiResponse response, DsStorageApi dsAPI,
                                                       String origin, AtomicInteger totalRecordsLoaded,
-                                                     AtomicInteger xipCollections) throws ApiException {
+                                                     AtomicInteger xipCollectionsCount, AtomicInteger manRelRefNot2Count) throws ApiException {
         for (OaiRecord  oaiRecord : response.getRecords()) {
             totalRecordsLoaded.getAndAdd(1);
             String storageId=origin+":"+oaiRecord.getId();
-            if (oaiRecord.getMetadata().contains("<xip:Collection")){
+            if (skipPvicaXip(oaiRecord.getMetadata())){
                 // XIP:Collections do not provide any needed metadata. Therefore, they are not added to ds-storage.
-                xipCollections.getAndAdd(1);
-            } else if (oaiRecord.isDeleted()) { //mark for delete
+                xipCollectionsCount.getAndAdd(1);                                                
+            }
+            else if (skipManRefRefNot2(oaiRecord.getMetadata())) {
+                manRelRefNot2Count.getAndAdd(1);                                
+            }
+            else if (oaiRecord.isDeleted()) { //mark for delete
                 dsAPI.markRecordForDelete(storageId);
             } else { //Create or update
-                addOrUpdateRecord(oaiRecord, storageId, origin, dsAPI);
+                String parent=getPvicaParent(oaiRecord.getMetadata());
+                if (parent != null) {
+                    parent=origin+":"+parent;
+                }                
```

Comment:
Maybe add origin as input for getPvicaParent and combine origin and parent in the method?